klient/wait/main_test.go

```diff
@@ -0,0 +1,143 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package wait
+
+import (
+	"context"
+	appsv1 "k8s.io/api/apps/v1"
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"log"
+	"os"
+	"sigs.k8s.io/e2e-framework/klient/internal/testutil"
+	"sigs.k8s.io/e2e-framework/klient/k8s/resources"
+	"sync"
+	"testing"
+
+	"k8s.io/client-go/rest"
+)
+
+var (
+	tc *testutil.TestCluster
+	cfg        *rest.Config
+	resourceManager *resources.Resources
+	namespace = "wait-test"
+	resourceManagerOnce sync.Once
+)
+
+func TestMain(m *testing.M) {
+	tc = testutil.SetupTestCluster("")
+	cfg = tc.RESTConfig
+	setup()
+	exitCode := m.Run()
+	tearDown()
+	os.Exit(exitCode)
+}
+
+func setup() {
+	createNamespace()
+}
+
+func getResourceManager() *resources.Resources {
+	resourceManagerOnce.Do(func() {
+		resourceMgr, err := resources.New(cfg)
+		if err != nil {
+			log.Fatalln("failed to create a resource manager instance", err)
+		}
+		resourceManager = resourceMgr.WithNamespace(namespace)
+	})
+	return resourceManager
+}
+
+func tearDown() {
+	deleteNamespace()
+	tc.DestroyTestCluster()
+}
+
+func deleteNamespace() {
+	namespace := &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: namespace}}
+	_ = getResourceManager().Delete(context.TODO(), namespace)
```

Comment:
I understand why error is not handled (because it's used during tear down), but I would suggest return the error here and ignore it when it's used (in `tearDown`)

Reply:
Let me update this.